### PR TITLE
Support modules that point to the same path in manifest.mf

### PR DESCRIPTION
**Pull request by tal-sapan**
Note: this is currently only relevant in the assemble scenario.

### Checklist
- [X] Code compiles correctly
- [X] Relevant tests were added (unit / contract / integration)
- [ ] Relevant logs were added
- [X] Formatting and linting run locally successfully
- [X] All tests pass
- [ ] UA review
- [ ] Design is documented
- [ ] Extended the README / documentation, if necessary
- [ ] Open source is approved

### CODE REVIEW

File: internal/artifacts/manifest.go [hunk 1]
```diff
@@ -63,9 +63,45 @@ func setManifestDesc(source dir.ISourceModule, ep dir.ITargetArtifacts, targetPa
...
+	mergedEntries := make([]entry, 0)
+	modules := make(map[string]entry)
+	// To keep a consistent sort order for the map entries we must keep another data structure (slice of keys by order of addition here)
+	pathsOrder := make([]string, 0)
+
+	// Add each non-module entry to mergedEntries. Add module entries to modules.
+	for index, entry := range entries {
+		if entry.EntryType != moduleEntry {
+			mergedEntries = append(mergedEntries, entry)
+			continue
```
Comment (allaVolkov) [inline internal/artifacts/manifest.go]: please use if else instead

File: internal/artifacts/manifest_test.go [hunk 2]
```diff
@@ -8,6 +8,7 @@ import (
 
 	"github.com/SAP/cloud-mta/mta"
 	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/ginkgo/extensions/table"
```
Comment (allaVolkov) [inline internal/artifacts/manifest_test.go]: imports sequence

File: internal/artifacts/manifest.go (hunk 1 above)
Comment (tal-sapan) [reply internal/artifacts/manifest.go]: Done

**Comment by tal-sapan** [comment]
It says the coverage fell but all the new code is covered, and I don't want to add tests for things that weren't changed in this PR

**Pull request merged**